### ISSUE TITLE
MIgrating to New Architecture

### ISSUE THREAD

**Issue by salisuabubakar** [grahammendick/navigation#716]
Hi,
Am having some challenges migrating my app to the new artchitecture. When i try to build using xcode i get this error 

No matching constructor for initialization of 'facebook::react::NVTabBarItemState'.

Below is the file that i get the error on:

```
#pragma once

#include "NVTabBarItemState.h"
#include <react/renderer/components/view/ConcreteViewShadowNode.h>
#include <react/renderer/components/navigationreactnative/EventEmitters.h>
#include <react/renderer/components/navigationreactnative/Props.h>
#include <react/renderer/imagemanager/ImageManager.h>

namespace facebook {
namespace react {

JSI_EXPORT extern const char NVTabBarItemComponentName[];

class JSI_EXPORT NVTabBarItemShadowNode final: public ConcreteViewShadowNode<
  NVTabBarItemComponentName,
  NVTabBarItemProps,
  NVTabBarItemEventEmitter,
  NVTabBarItemState
> {

public:
  using ConcreteViewShadowNode::ConcreteViewShadowNode;

  void setImageManager(const SharedImageManager &imageManager);

  static NVTabBarItemState initialStateData(
                                           ShadowNodeFragment const &fragment,
                                           ShadowNodeFamilyFragment const &familyFragment,
                                           ComponentDescriptor const &componentDescriptor) {
    auto imageSource = ImageSource{ImageSource::Type::Invalid};
    return {
      imageSource,
      {imageSource, nullptr}};
  }

#pragma mark - LayoutableShadowNode

  Size measureContent(
                      LayoutContext const &layoutContext,
                      LayoutConstraints const &layoutConstraints) const override;
  void layout(LayoutContext layoutContext) override;

private:
  void updateStateIfNeeded();

  ImageSource getImageSource() const;

  SharedImageManager imageManager_;
};

} // namespace react
} // namespace facebook

```

Kindly check if its something you can help me resolve. 

Thanks

**Comment by salisuabubakar** [comment]
Hi,
Thanks for the quick response. I have tried the sample fabri new architecture but i have this error stated above when i try to run my app.

**Comment by grahammendick** [comment]
Can you successfully run the [fabric sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/fabric) from this repo on iOS and Android?

**Comment by salisuabubakar** [comment]
![image](https://github.com/grahammendick/navigation/assets/6886556/9eab6114-d691-4510-aaa1-1846b12b9237)

**Comment by grahammendick** [comment]
I want to know if you get an error when you run the fabric sample in this repo.
Please can you clone this repository, run the [fabric sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/fabric) and let me know what happens.

**Comment by salisuabubakar** [comment]
Hi, Yes i have run it before and i didnt have any issue. Thanks

**Comment by grahammendick** [comment]
Ok, then you've probably not upgraded your project to the new architecture quite right.
Why don't you create a brand new project on the new architecture, make sure it runs and then copy your files from the old project to the new one?

**Comment by salisuabubakar** [comment]
Sure. I think thats a good idea. Thanks soo much. Really appreciate your time and support. Am very proud to say all the apps i built using the navigation router works perfect. Its hard to deferentiate between that of a native app. Thanks for the wonderful library one again.

**Comment by grahammendick** [comment]
Oh thank you. I'm always happy to help and it's so nice to hear that feedback. Please tell all your friends!

**Comment by grahammendick** [comment]
Are your apps public? I'd like to see what you've built

**Comment by salisuabubakar** [comment]
Yes sure. Please search for Oil City Radio 95.9 on app store.

**Comment by salisuabubakar** [comment]
HI, am still getting the same error. Can you please check. Thanks

**Comment by grahammendick** [comment]
Oh yea, I just tried it with the last react-native and get the same error. I'll investigate. Thanks for letting me know

**Comment by salisuabubakar** [comment]
Sure. Thanks soo much. I really need it.

**Comment by grahammendick** [comment]
Think I've [found the fix](https://github.com/software-mansion/react-native-svg/pull/2079/files). If you want to try it out you need to put it in `NVBarButtonShadowNode.h`, `NVNavigationBarShadowNode.h` and `NVTabBarItemShadowNode.h`.

**Comment by grahammendick** [comment]
I just published the fix in navigation-react-native v9.4.2 and [thanked you in the release notes](https://github.com/grahammendick/navigation/releases/tag/v9.4.2-NavigationReactNative)

**Comment by salisuabubakar** [comment]
Thanks soo much. i have update navigation-react-native to v9.4.2 and its working.
